labcontrol/gui/static/js/labcontrol.js

```diff
@@ -742,3 +742,36 @@ function createHeatmap(
     });
   });
 }
+
+/**
+ *
+ * Given a query string and an array of strings to search through, returns a
+ * subset of the array containing only strings that contain the query string.
+ *
+ * This search is case insensitive, so "abc" will match with "ABCDEF".
+ *
+ * Note that if queryString is empty (i.e. "") then this won't return any
+ * matches, even if "" is present in stringArray for some reason (since
+ * stringArray should be a list of sample IDs, this should never be the case).
+ *
+ * This function is loosely based on textFilterFeatures() in Qurro:
+ * https://github.com/biocore/qurro/blob/3f4650fb677753e978d971b06794d4790b051d30/qurro/support_files/js/feature_computation.js#L29
+ *
+ * @param {string} queryString The string that will be searched for
+ * @param {array} stringArray Collection of strings to check against the query
+ * @returns {array} Collection of strings in stringArray that include the query
+ *
+ */
+function getSubstringMatches(queryString, stringArray) {
+  if (queryString.length === 0 || stringArray.length === []) {
+    return [];
+  }
+  var queryStringLowerCase = queryString.toLowerCase();
```

Comment:
I'm ok with that (so long as we're ok with only supporting ES6-capable browsers). To clarify, do you just mean using `let` instead of `var` from here on, or actively porting the old code to use `let`?

`grep -ri "var " labcontrol/gui/static/js/ labcontrol/gui/js_tests/*.js | wc -l` finds 171 hits and `grep -ri "let " labcontrol/gui/static/js/ labcontrol/gui/js_tests/*.js | wc -l` finds 28 hits, so the latter option might take some time :)